src/plasmapy/particles/ionization_state_collection.py

```diff
@@ -476,15 +476,15 @@ def ionic_fractions(  # noqa: C901, PLR0912, PLR0915
             # The particles whose ionization states are to be recorded
             # should be elements or isotopes but not ions or neutrals.
 
-            for key in particles:
-                is_element = particles[key].is_category("element")
-                has_charge_info = particles[key].is_category(
+            for particle in particles:  # noqa: PLC0206
+                is_element = particles[particle].is_category("element")
+                has_charge_info = particles[particle].is_category(
```

Comment:
In retrospect, my choice of `key` as a variable name was perhaps not the most descriptive. This will be better, at least.